pandas-stubs/core/arrays/base.pyi

```diff
@@ -43,10 +43,7 @@ class ExtensionArray:
     def unique(self): ...
     def searchsorted(self, value, side: str = ..., sorter=...): ...
     def factorize(
-        self,
-        # Not actually positional-only, used to handle deprecations in 1.5.0
-        *,
-        use_na_sentinel: bool = ...,
+        self, use_na_sentinel: bool = ...
```

Comment:
Is this correct?  In pandas 1.5.x `na_sentinel` is still there, hence the positional only usage.

Reply:
I agree it would be better to keep it as keyword-only. `na_sentinel` is deprecated in 1.5 (and therefore removed from the stubs) but `use_na_sentinel` is the last argument so we would need to make it keyword-only to encourage correct and future-proof usage.